pywbem_mock/_inmemoryrepository.py

```diff
@@ -0,0 +1,279 @@
+#
+# (C) Copyright 2020 InovaDevelopment.comn
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+#
+# Author: Karl  Schopmeyer <inovadevelopment.com>
+#
+
+"""
+The pywbem_mock module provides a standard interface to a CIM repository that
+stores collections of :class:`~pywbem.CIMClass`, :class:`~pywbem.CIMInstance`,
+and :class:`~pywbem.CIMQualifierDeclaration` as Python asbstract classes in the
+module :py:mod:`pywbem_mock/_baserepository.py`.
+
+The pywbem_mock inmemory CIM repository implements a repository derived from
+:py:mod:`pywbem_mock/_baserepository.py` that stores the CIM objects only in
+memory. This CIMrepository is created each time a
+:class:`~pywbem_mock.FakedWBEMConnection` is constructed and destroyed each
+time it is destroyed.
+
+.. code-block:: python
+  Example:
+  repo = InMemoryRepository()                      # create the repo
+  repo.add_namespace("root/cimv2")                 # add a namespace
+  class_store = .repo.get_class_store("root/cimv2") # get class obj store
+  test_class = CIMClass('CIM_Blah', ...)           # create a class
+  class_store.add(test_class)                      # add to xxxrepo classes
+  if 'CIM_Blah' in class_store:                    # test if class exists
+      klass = class_store.get('CIM_Blah;)          # get the class
```

Comment:
This code block fails in Sphinx. To fix it, have a blank line after the code-block directive and indent the lines in the code block to have 3 or 4 spaces.
Also, probably the "Example:" line should be outside of the code block.

Reply:
Whoops.  Think I moved it a bit after running builddoc. DONE.  Also changed example in _baserepository despite fact that this one is not in documentation.

Reply:
This one was not done in the last commit.

Reply:
DONE in new commit.